### PR TITLE
Add log for debug in sanity check. 

### PR DESCRIPTION
**Pull request by yutongzhang-microsoft**
### Description of PR

A typeError sometimes occurs in common/plugins/sanity_check/__init__.py:206, but existing log can not help us find out the reason of this failure. So add some logs for debug. 

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012

#### What is the motivation for this PR?
A typeError sometimes occurs in common/plugins/sanity_check/__init__.py:206, but existing log can not help us find out the reason of this failure. So add some logs for debug. 

#### How did you do it?
Add some dubug logs.

### CODE REVIEW

File: tests/common/plugins/sanity_check/__init__.py [hunk 1]
```diff
@@ -94,10 +94,12 @@ def do_checks(request, check_items, *args, **kwargs):
     for item in check_items:
         check_fixture = request.getfixturevalue(item)
         results = check_fixture(*args, **kwargs)
+        logger.debug("check results of each item {}".format(results))
```
Comment (wangxin) [inline tests/common/plugins/sanity_check/__init__.py:97]: This is unnecessary.

File: tests/common/plugins/sanity_check/__init__.py [hunk 2]
```diff
@@ -94,10 +94,12 @@ def do_checks(request, check_items, *args, **kwargs):
...
         if results and isinstance(results, list):
             check_results.extend(results)
         elif results:
             check_results.append(results)
+    logger.debug("check_results {}".format(check_results))
```
Comment (wangxin) [inline tests/common/plugins/sanity_check/__init__.py:102]: This is unnecessary. The `check_results` returned by `do_checks` is already logged in line 230 and 250.
Too much redundant information will make it harder to search through the test log to find useful information.

**Review by wangxin — CHANGES_REQUESTED** [review]
I think the logs added in this PR have already been logged. They won't be able to give any new information for debugging.

File: tests/common/plugins/sanity_check/__init__.py (hunk 2 above)
Comment (yutongzhang-microsoft) [reply tests/common/plugins/sanity_check/__init__.py]: I delete this.

File: tests/common/plugins/sanity_check/__init__.py (hunk 1 above)
Comment (yutongzhang-microsoft) [reply tests/common/plugins/sanity_check/__init__.py:97]: I think it's necessary. This log shows the form of the variable `results` before it is added into check_results.

Comment (wangxin) [reply tests/common/plugins/sanity_check/__init__.py:97]: I got your point. Agree.

File: tests/common/plugins/sanity_check/checks.py [hunk 3]
```diff
@@ -75,6 +75,7 @@ def _find_down_ports(dut, phy_interfaces, ip_interfaces):
 def check_interfaces(duthosts):
     def _check(*args, **kwargs):
         result = parallel_run(_check_interfaces_on_dut, args, kwargs, duthosts.frontend_nodes, timeout=600)
+        logger.debug("Result of check_interfaces: {}".format(result))
```
Comment (wangxin) [inline tests/common/plugins/sanity_check/checks.py:78]: I think the logs added in this file are unnecessary. They should can be logged by the change you made to `sanity_check/__init__.py`, right?

Comment (yutongzhang-microsoft) [reply tests/common/plugins/sanity_check/checks.py]: Yes, you are right, I delete these.

**Review by wangxin — APPROVED** [review]

**Pull request merged**